util-core/src/main/scala/com/twitter/util/Event.scala

```diff
@@ -430,9 +430,7 @@ object Witness {
   /**
    * Create a [[Witness]] from a [[Function1]].
    */
-  def apply[T](f: T => Unit): Witness[T] = new Witness[T] {
-    def notify(t: T): Unit = f(t)
-  }
+  def apply[T](f: T => Unit): Witness[T] = fromFunction(f)
```

Comment:
could you elaborate on what the problem is?  why can't scala do this?  is this a bug in 2.12.0-M4, or is this expected behavior?